### ISSUE TITLE
roles index: _id_ dup key: { : "admin" }

### ISSUE THREAD

**Issue by guncebektas** [jamauro/soft-delete#2]
I'm getting this error on v.0.2.1 the reproduction can be found in https://github.com/guncebektas/saas-boilerplate.
Removing this package solves the issue.

```
 MongoServerError: E11000 duplicate key error collection: saas.roles index: _id_ dup key: { : "admin" }
I20240924-23:07:05.170(3)?     at /Users/gunce/.meteor/packages/npm-mongo/.4.17.4.1wekbp8.zx5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/src/operations/insert.ts:85:25
I20240924-23:07:05.170(3)?     at /Users/gunce/.meteor/packages/npm-mongo/.4.17.4.1wekbp8.zx5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/src/cmap/connection_pool.ts:569:13
I20240924-23:07:05.170(3)?     at /Users/gunce/.meteor/packages/npm-mongo/.4.17.4.1wekbp8.zx5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/src/sdam/server.ts:363:13
I20240924-23:07:05.170(3)?     at handleOperationResult (/Users/gunce/.meteor/packages/npm-mongo/.4.17.4.1wekbp8.zx5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/src/sdam/server.ts:496:14)
I20240924-23:07:05.170(3)?     at Connection.onMessage (/Users/gunce/.meteor/packages/npm-mongo/.4.17.4.1wekbp8.zx5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/src/cmap/connection.ts:461:5)
I20240924-23:07:05.170(3)?     at MessageStream.<anonymous> (/Users/gunce/.meteor/packages/npm-mongo/.4.17.4.1wekbp8.zx5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/src/cmap/connection.ts:241:56)
I20240924-23:07:05.170(3)?     at MessageStream.emit (node:events:519:28)
I20240924-23:07:05.170(3)?     at MessageStream.emit (node:domain:488:12)
I20240924-23:07:05.170(3)?     at processIncomingData (/Users/gunce/.meteor/packages/npm-mongo/.4.17.4.1wekbp8.zx5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/src/cmap/message_stream.ts:188:12)
I20240924-23:07:05.170(3)?     at MessageStream._write (/Users/gunce/.meteor/packages/npm-mongo/.4.17.4.1wekbp8.zx5e++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/mongodb/src/cmap/message_stream.ts:69:5)
I20240924-23:07:05.170(3)?     at writeOrBuffer (node:internal/streams/writable:570:12)
I20240924-23:07:05.170(3)?     at _write (node:internal/streams/writable:499:10)
I20240924-23:07:05.170(3)?     at MessageStream.Writable.write (node:internal/streams/writable:508:10)
I20240924-23:07:05.170(3)?     at Socket.ondata (node:internal/streams/readable:1007:22)
I20240924-23:07:05.170(3)?     at Socket.emit (node:events:519:28)
I20240924-23:07:05.170(3)?     at Socket.emit (node:domain:488:12)
```

**Comment by jamauro** [comment]
Thanks for reporting. I plan to address this soon.

**Comment by jamauro** [comment]
This has been fixed in `v0.3.0` 🎉 . You can now [exclude](https://github.com/jamauro/soft-delete?tab=readme-ov-file#configuring-optional) specific collections from using soft delete. The collections created by the roles package are excluded by default.

**Comment by guncebektas** [comment]
It's working and I don't have a problem with my use case but old data that don't have "delete: false" might be an issue for real-world scenarios. 

I think future readers should pay attention to this point and migrate their data. 

Thank you @jamauro for this great package.
